pmd-core/src/main/java/net/sourceforge/pmd/lang/ParserOptions.java

```diff
@@ -4,19 +4,38 @@
 
 package net.sourceforge.pmd.lang;
 
+import java.util.Locale;
+import java.util.Objects;
+
+import net.sourceforge.pmd.properties.AbstractPropertySource;
+import net.sourceforge.pmd.properties.PropertyDescriptor;
+
 /**
  * Represents a set of configuration options for a {@link Parser}. For each
  * unique combination of ParserOptions a Parser will be used to create an AST.
  * Therefore, implementations must implement {@link Object#equals(Object)} and
  * {@link Object#hashCode()}.
  */
-public class ParserOptions {
+public class ParserOptions extends AbstractPropertySource {
```

Comment:
I think it would be a better idea to hide the property source inside the parser options instead of using inheritance. This would avoid eg splitting your equals method with that `super` call. I'd like to remove AbstractPropertySource eventually, the inheritance is really causing problems.